packages/@apphosting/adapter-nextjs/src/bin/build.ts

```diff
@@ -4,16 +4,27 @@ import {
   build,
   populateOutputBundleOptions,
   generateOutputDirectory,
+  DEFAULT_COMMAND,
 } from "../utils.js";
-
 import { join } from "path";
 
-const cwd = process.cwd();
+const root = process.cwd();
+
+let projectRoot = root;
+if (process.env.MONOREPO_PROJECT && process.env.FIREBASE_APP_DIRECTORY) {
+  projectRoot = projectRoot.concat("/", process.env.FIREBASE_APP_DIRECTORY);
+}
+
+// Determine which build runner to use
+let cmd = DEFAULT_COMMAND;
+if (process.env.MONOREPO_COMMAND) {
+  cmd = process.env.MONOREPO_COMMAND;
```

Comment:
Is this generalizable to all monorepos?

Seems like `turborepo` uses `turborepo run ...` which seems appropriate, lerna uses `npx lerna run ...`

Reply:
Yes, they mirror the syntax of npm scripts (`npm run build`, `nx run build`, `turbo run build`, etc.)

Reply:
However some have specific syntax, e.g. Nx allows specifying `nx run [project]:build:[environment]`.

Reply:
This seems like an argument for offloading the build command syntax to a shared library in the adapters